src/shims/windows/handle.rs

```diff
@@ -0,0 +1,162 @@
+use rustc_target::abi::HasDataLayout;
+use std::mem::variant_count;
+
+use crate::*;
+
+/// A Windows `HANDLE` that represents a resource instead of being null or a pseudohandle.
+///
+/// This is a seperate type from [`Handle`] to simplify the packing and unpacking code.
+#[derive(Clone, Copy)]
+enum RealHandle {
+    Thread(ThreadId),
+}
+
+impl RealHandle {
+    const USABLE_BITS: u32 = 31;
+
+    const THREAD_DISCRIMINANT: u32 = 1;
+
+    fn discriminant(self) -> u32 {
+        match self {
+            // can't use zero here because all zero handle is invalid
```

Comment:
This comment should probably go up to where `const THREAD_DISCRIMINANT` is defined.